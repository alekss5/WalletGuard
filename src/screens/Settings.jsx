import { useCallback } from 'react';
import { StyleSheet, ScrollView, View, Dimensions } from 'react-native';

import SettingsItem from '../components/SettingsItem';
import BackgroundColorContainer from '../components/UI/BackgroundColorContainer';
import BackgroundImageContainer from '../components/UI/BackgroundImageContainer';
import CustomText from '../components/UI/CustomText';
import CustomButton from '../components/UI/CustomButton';
import { lightVibration } from '../utils/vibrationPaterns';
import { useSelector } from 'react-redux';
import { selectEmail } from '../redux/selectors/personalInf';
import { SafeAreaView } from 'react-native-safe-area-context';

const { width } = Dimensions.get('window');
const isSmallDevice = width < 370;
export default function Settings({ navigation }) {
    const email = useSelector(selectEmail)
    const isLoggedIn = email && email.trim() !== ""

    const handleNavigation = useCallback((route) => {
        lightVibration();
        navigation.navigate(route);
    }, [navigation]);

    const navigateToHelp = () => {
        lightVibration();
        navigation.navigate('CommingSoon');

    }
    const navigateToInstagram = () => {
        lightVibration();
        navigation.navigate('CommingSoon');

    }
    const logOut = () => {

    }

    return (
        <BackgroundColorContainer>
            <BackgroundImageContainer>
                <View style={styles.container}>
                    {/* <CustomButton
                        onPress={() => handleNavigation('Upgrade')}
                        style={styles.upgradeButton}
                        textStyle={styles.upgradeButtonText}
                    >
                        Upgrade to premium
                    </CustomButton> */}
                    <SafeAreaView>

                    <ScrollView>
                        <Section title="System">
                            <SettingsItem onPress={() => handleNavigation('PersonalInformation')} icon="person" text="Personal Information" />
                            <SettingsItem onPress={() => handleNavigation('StartDate')} icon="calendar" text="Start Date" />
                            <SettingsItem onPress={() => handleNavigation('CurrencyList')} icon="logo-euro" text="Currency" />
                            <SettingsItem onPress={() => handleNavigation('Subscriptions')} icon="calendar-number-outline" text="Subscriptions" />
                            <SettingsItem onPress={() => handleNavigation('Budget')} icon="wallet-outline" text="Budget" />
                        </Section>
                        <Section title="Preferences">
                            <SettingsItem onPress={() => handleNavigation('Theme')} icon="phone-portrait" text="Theme" />
                            <SettingsItem onPress={() => handleNavigation('Notifications')} icon="notifications" text="Notifications" />
                            <SettingsItem onPress={() => handleNavigation('Privacy')} icon="lock-open-outline" text="Privacy" />
                        </Section>
                        <Section title="Support & Feedback">
                            <SettingsItem onPress={navigateToHelp} icon="help-circle-outline" text="Help" />
                            <SettingsItem onPress={navigateToInstagram} icon="logo-instagram" text="Instagram" />
                        </Section>
                        {/* <Section title="Account">
                            {isLoggedIn ? <SettingsItem onPress={logOut} icon="log-out-outline" text="Log out" /> : <>
                                <SettingsItem onPress={() => handleNavigation('Register')} icon="create-outline" text="Create account" />
                                <SettingsItem onPress={() => handleNavigation('Login')} icon="log-in-outline" text="Log in" />
                            </>}
                        </Section> */}
                    </ScrollView>
                    </SafeAreaView>

                </View>
            </BackgroundImageContainer>
        </BackgroundColorContainer>
    );
}

const Section = ({ title, children }) => (
    <>
        <CustomText style={styles.headerText}>{title}</CustomText>
        {children}
    </>
);

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: '90%',
        alignSelf: 'center',
    },
    upgradeButton: {
        width: '60%',
        height: 50,
        alignSelf: 'center',
        marginVertical: 40,
        marginTop: 80,
        ...(isSmallDevice && { marginTop: 50, }),
    },
    upgradeButtonText: {
        fontWeight: '700',
    },
    headerText: {
        marginTop: 20,
        fontSize: 16,
    },
    divider: {
        height: 1,
        width: '100%',
        alignSelf: 'center',
        marginVertical: 5,
    },
});
