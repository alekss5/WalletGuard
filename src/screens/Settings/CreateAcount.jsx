import { StyleSheet, Text, View } from 'react-native'
export default function CreateAcount() {
  return (
    <View>
      <Text>CreateAcount</Text>
    </View>
  )
}

const styles = StyleSheet.create({})