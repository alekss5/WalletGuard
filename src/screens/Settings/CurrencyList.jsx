import React, { useState, useMemo, useCallback } from 'react';
import { View, FlatList, StyleSheet, TouchableOpacity } from 'react-native';
import currencyCodes from 'currency-codes';
import getSymbolFromCurrency from 'currency-symbol-map';
import { useTheme } from 'react-native-paper';
import { useDispatch, useSelector } from 'react-redux';
import { selectCurrency, setCurrency } from '../../redux/budgetReducer';
import BackgroundColorContainer from '../../components/UI/BackgroundColorContainer';
import CustomText from '../../components/UI/CustomText';
import CustomInput from '../../components/UI/CustomInput';

const CurrencyList = ({ navigation }) => {
  const { colors } = useTheme();
  const dispatch = useDispatch();

  const selectedCurrency = useSelector(selectCurrency);
  const [searchQuery, setSearchQuery] = useState('');

  const filteredCurrencies = useMemo(() => {
    if (!searchQuery) return currencyCodes.data;
    return currencyCodes.data.filter(currency =>
      currency.currency.toLowerCase().includes(searchQuery.toLowerCase()) ||
      currency.code.toLowerCase().includes(searchQuery.toLowerCase()) ||
      currency.countries.some(country => country.toLowerCase().includes(searchQuery.toLowerCase()))
    );
  }, [searchQuery]);

  const onChangeSearch = useCallback((query) => {
    setSearchQuery(query);
  }, []);

  const handlePress = useCallback((item) => {
    const symbol = getSymbolFromCurrency(item.code);
 
    dispatch(setCurrency(symbol));
    navigation.goBack();
  }, [dispatch, navigation]);

  return (
    <BackgroundColorContainer>
      <CustomInput
        placeholder="Search"
        onChangeText={onChangeSearch}
        value={searchQuery}
        style={styles.searchbar}
      />

      <FlatList
        data={filteredCurrencies}
        keyExtractor={(item) => item.code}
        renderItem={({ item }) => {
          const itemBackgroundColor = getSymbolFromCurrency(item.code) === selectedCurrency ? colors.accent : colors.background;
          return (
            <TouchableOpacity onPress={() => handlePress(item)}>
              <View style={[styles.item, { backgroundColor: itemBackgroundColor }]}>
                <CustomText style={styles.text}>
                  {item.code} - {item.currency} ({getSymbolFromCurrency(item.code)}) - {item.countries.join(', ')}
                </CustomText>
              </View>
            </TouchableOpacity>
          );
        }}
      />
    </BackgroundColorContainer>
  );
};

const styles = StyleSheet.create({
  searchbar: {
    marginBottom: 10,
    marginTop: 10,
    width: '95%',
    alignSelf: 'center',
  },
  item: {
    width: '98%',
    borderRadius: 10,
    padding: 10,
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
  },
  text: {
    fontSize: 16,
  },
});

export default CurrencyList;
