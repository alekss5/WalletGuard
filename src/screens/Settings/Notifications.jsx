import { StyleSheet, View } from 'react-native';
import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import BackgroundColorContainer from '../../components/UI/BackgroundColorContainer';
import SettingsItem from '../../components/SettingsItem';
import {
  toggleNotification
} from '../../redux/uiReducer';
import { selectDailyNotification,
  selectOverBudgetNotification,
  selectWeeklyNotification,
  selectSubscNotification,
  selectMonthlyRefreshNotification, } from '../../redux/selectors/ui';

export default function Notifications() {
  const dispatch = useDispatch();

  const dailyNotification = useSelector(selectDailyNotification);
  const overBudgetNotification = useSelector(selectOverBudgetNotification);
  const weeklyNotification = useSelector(selectWeeklyNotification);
  const subscNotification = useSelector(selectSubscNotification);
  const monthlyRefreshNotification = useSelector(selectMonthlyRefreshNotification);

  const handleToggle = (notificationType) => {
    dispatch(toggleNotification({ notificationType }));
  };

  return (
    <BackgroundColorContainer>
      <View style={{ width: '90%', alignSelf: 'center', paddingTop: 10 }}>
        <SettingsItem
          text="Daily Notifications"
          icon="notifications-circle"
          isSwitchOn={dailyNotification}
          onToggleSwitch={() => handleToggle('dailyNotification')}
        />
        <SettingsItem
          text="Over Budget"
          icon="trending-down"
          isSwitchOn={overBudgetNotification}
          onToggleSwitch={() => handleToggle('overBudgetNotification')}
        />
        <SettingsItem
          text="Weekly Notifications"
          icon="calendar-sharp"
          isSwitchOn={weeklyNotification}
          onToggleSwitch={() => handleToggle('weeklyNotification')}
        />
        <SettingsItem
          text="Subscriptions"
          icon="calendar-number-outline"
          isSwitchOn={subscNotification}
          onToggleSwitch={() => handleToggle('subscriptions')}
        />
        <SettingsItem
          text="Monthly refresh"
          icon="refresh"
          isSwitchOn={monthlyRefreshNotification}
          onToggleSwitch={() => handleToggle('monthlyRefresh')}
        />
      </View>
    </BackgroundColorContainer>
  );
}

const styles = StyleSheet.create({});
