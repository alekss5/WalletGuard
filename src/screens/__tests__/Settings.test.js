import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Settings from '../Settings';
import { lightVibration } from '../../utils/vibrationPaterns';

// Mock navigation and vibration
const mockNavigate = jest.fn();
jest.mock('../../utils/vibrationPaterns', () => ({
  lightVibration: jest.fn(),
}));

describe('Settings Component', () => {
  const navigation = { navigate: mockNavigate };

  it('renders correctly', () => {
    const { toJSON } = render(<Settings navigation={navigation} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('navigates to correct routes when buttons are pressed', () => {
    const { getByText } = render(<Settings navigation={navigation} />);

    fireEvent.press(getByText('Upgrade to premium'));
    expect(mockNavigate).toHaveBeenCalledWith('CommingSoon');
    expect(lightVibration).toHaveBeenCalled();

    fireEvent.press(getByText('Personal Information'));
    expect(mockNavigate).toHaveBeenCalledWith('PersonalInformation');
    expect(lightVibration).toHaveBeenCalled();

  });
});
