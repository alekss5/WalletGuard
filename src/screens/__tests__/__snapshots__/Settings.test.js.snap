// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Component renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "testUri": "../../../assets/bg.jpeg",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          {
            "opacity": 0.1,
          },
        ]
      }
    />
    <View
      style={
        {
          "alignSelf": "center",
          "flex": 1,
          "width": "90%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "rgba(255, 251, 254, 1)",
            "borderRadius": 10,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginTop": 80,
            "marginVertical": 40,
            "opacity": 1,
            "padding": 10,
            "width": "60%",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": undefined,
                "fontSize": 18,
              },
              {
                "fontWeight": "700",
              },
            ]
          }
        >
          Upgrade to premium
        </Text>
      </View>
      <RCTScrollView>
        <View>
          <Text
            style={
              [
                {
                  "color": undefined,
                },
                {
                  "fontSize": 16,
                  "marginTop": 20,
                },
              ]
            }
          >
            System
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="person"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Personal Information
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="calendar"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Start Date
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="logo-euro"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Currency
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="calendar-number-outline"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Subscriptions
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="wallet-outline"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Budget
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <Text
            style={
              [
                {
                  "color": undefined,
                },
                {
                  "fontSize": 16,
                  "marginTop": 20,
                },
              ]
            }
          >
            Preferences
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="phone-portrait"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Theme
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="notifications"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Notifications
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="lock-open-outline"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Privacy
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <Text
            style={
              [
                {
                  "color": undefined,
                },
                {
                  "fontSize": 16,
                  "marginTop": 20,
                },
              ]
            }
          >
            Support & Feedback
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="help-circle-outline"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Help
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="logo-instagram"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Instagram
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <Text
            style={
              [
                {
                  "color": undefined,
                },
                {
                  "fontSize": 16,
                  "marginTop": 20,
                },
              ]
            }
          >
            Account
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="create-outline"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Create account
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginVertical": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <Ionicons
                  name="log-in-outline"
                  size={30}
                  style={
                    {
                      "padding": 5,
                    }
                  }
                  testID="settings-icon"
                />
                <Text
                  style={
                    [
                      {
                        "color": undefined,
                      },
                      {
                        "fontSize": 20,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  Log in
                </Text>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "rgba(202, 196, 208, 1)",
                    "height": 0.5,
                  },
                  undefined,
                  false,
                  false,
                  [
                    {
                      "marginTop": 10,
                    },
                    {
                      "backgroundColor": undefined,
                    },
                  ],
                ]
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
